searx/webapp.py

```diff
@@ -866,6 +888,42 @@ def autocompleter():
     return Response(suggestions, mimetype=mimetype)
 
 
+@app.route('/favicon', methods=['GET'])
+def favicon_proxy():
+    """Return proxied favicon results"""
+    url = request.args.get('q')
+
+    # malformed request
+    if not url:
+        return '', 400
+
+    # malformed request / does not have authorisation
+    if not is_hmac_of(settings['server']['secret_key'], url.encode(), request.args.get('h', '')):
+        return '', 400
+
+    resolver = request.preferences.get_value('favicon_resolver')
+
+    # check if the favicon resolver is valid
+    if not resolver or resolver not in favicon_backends:
+        return '', 400
+
+    # parse query
+    raw_text_query = RawTextQuery(url, [])
```

Comment:
I don't think using a `RawTextQuery` makes sense here since we're dealing with an fqdn, not a user query input?